src/Compilers/CSharp/Test/Semantic/Semantics/RefFieldTests.cs

```diff
@@ -12592,5 +12592,1268 @@ public void ScopedReserved_Alias_Escaped()
+        {
+            var sourceA =
+@"using System.Diagnostics.CodeAnalysis;
+public struct S<T>
+{
+    private T _t;
+    public ref T F1() => throw null;
+    [UnscopedRef] public ref T F2() => ref _t;
+}";
+            var comp = CreateCompilation(new[] { sourceA, UnscopedRefAttributeDefinition });
+            comp.VerifyEmitDiagnostics();
+            var refA = AsReference(comp, useCompilationReference);
+
+            var sourceB =
+@"class Program
+{
+    static ref int F1()
+    {
+        var s = new S<int>();
+        return ref s.F1();
+    }
+    static ref int F2()
+    {
+        var s = new S<int>();
+        return ref s.F2(); // 1
+    }
+}";
+            comp = CreateCompilation(sourceB, references: new[] { refA });
+            // https://github.com/dotnet/roslyn/issues/62791: Missing errors.
+            comp.VerifyEmitDiagnostics();
+        }
+
+        [Fact]
+        public void UnscopedRefAttribute_Property_01()
+        {
+            var source =
+@"using System.Diagnostics.CodeAnalysis;
+struct S1
+{
+    private int _field;
+    public ref int Property1 => ref _field; // 1
+    [UnscopedRef] public ref int Property2 => ref _field;
+}";
+            var comp = CreateCompilation(new[] { source, UnscopedRefAttributeDefinition });
+            comp.VerifyDiagnostics(
+                // (5,37): error CS8170: Struct members cannot return 'this' or other instance members by reference
+                //     public ref int Property1 => ref _field; // 1
+                Diagnostic(ErrorCode.ERR_RefReturnStructThis, "_field").WithLocation(5, 37));
+        }
+
+        [CombinatorialData]
+        [Theory]
+        public void UnscopedRefAttribute_Property_02(bool useCompilationReference)
+        {
+            var sourceA =
+@"using System.Diagnostics.CodeAnalysis;
+public struct S<T>
+{
+    private T _t;
+    public ref T P1 => throw null;
+    [UnscopedRef] public ref T P2 => ref _t;
+}";
+            var comp = CreateCompilation(new[] { sourceA, UnscopedRefAttributeDefinition });
+            comp.VerifyEmitDiagnostics();
+            var refA = AsReference(comp, useCompilationReference);
+
+            var sourceB =
+@"class Program
+{
+    static ref int F1()
+    {
+        var s = new S<int>();
+        return ref s.P1;
+    }
+    static ref int F2()
+    {
+        var s = new S<int>();
+        return ref s.P2; // 1
+    }
+}";
+            comp = CreateCompilation(sourceB, references: new[] { refA });
+            // https://github.com/dotnet/roslyn/issues/62791: Missing errors.
+            comp.VerifyEmitDiagnostics();
+        }
+
+        [Fact]
+        public void UnscopedRefAttribute_Property_03()
+        {
+            var source =
+@"using System.Diagnostics.CodeAnalysis;
+struct S
+{
+    [UnscopedRef] object P1 { get; }
+    [UnscopedRef] object P2 { get; set; }
+    [UnscopedRef] object P3 { get; init; } // 1
+    object P5
+    {
+        [UnscopedRef] get;
```

Comment:
nit: Could add link to tracking issue
In addition to errors on `init`, I'd expect all these scenarios to be errors since only by-value returns and parameters. The "It can be applied to any `ref` which is implicitly scoped" rule should kick in.

Reply:
> The "It can be applied to any ref which is implicitly scoped" rule should kick in.

In these cases, the implicitly-scoped parameter is `this`.